### ISSUE TITLE
Display Bootstrap-like tooltips for menu items

### ISSUE THREAD

**Issue by jywarren** [publiclab/Leaflet.DistortableImage#104]
This may need to be upstream in Leaflet.Toolbar? https://github.com/leaflet/Leaflet.toolbar

R - toggle rotate/distort
L - lock
T - transparent
O - outline (pretty similar to transparent)

![image](https://user-images.githubusercontent.com/24359/43849016-5b74fbfc-9b02-11e8-81fe-0d70d51a6a8e.png)

@imvec

**Comment by roshniRam** [comment]
@jywarren Can I work on this?

**Comment by jywarren** [comment]
yes please, that would be great!

**Comment by jywarren** [comment]
Hi @roshniRam are you still interested in this at all? 

I think we may be able to remove the `tooltip` attribute here and just build in the Bootstrap-style tooltips into the `span`:

https://github.com/publiclab/Leaflet.DistortableImage/blob/90c977770658453d57233e1a150159cf7a38b1cb/src/edit/DistortableImage.EditToolbar.js#L14-L16

It would require calling the `tooltip` bootstrap javascript function here, too: 

https://getbootstrap.com/docs/3.3/javascript/#tooltips

**Comment by jywarren** [comment]
This has been marked as a good candidate for becoming a [first-timers-only issue](http://www.firsttimersonly.com) like [these](https://code.publiclab.org/#r=all), meaning that it's simple, self-contained, and with some extra formatting, could be a great entry point for a new contributor. If you're familiar enough with this code, please consider [reformatting or reposting it as a first-timers-only issue](https://publiclab.org/notes/warren/10-31-2016/create-a-welcoming-first-timers-only-issue-to-invite-new-software-contributors), and then ping @publiclab/reviewers to get it labelled. Or, if this is not your first time, try to solve it yourself!

****

**Comment by sashadev-sky** [comment]
Ah just saw this one after all my tooltip updates! I will open an issue for this soon to get one done and the workflow down then an FTO for the rest. Will use the nice tooltips from material kit pro to match the icons :)

**Comment by jywarren** [comment]
ok awesome. Not critical either, i just appreciate the tooltips you've put
in there, and at least in bootstrap, styling them is pretty easy and looks
great.

**Comment by rexagod** [comment]
Uploaded to GCI dashboard.

**Comment by SidharthBansal** [comment]
published

**Comment by neelesh17** [comment]
could i try this issue?

**Comment by sashadev-sky** [comment]
@neelesh17 yes would love your help on this! The one catch is that we should do it without pulling in an external library, so you you'll have to make them from scratch. (I find that more fun anyways) 👍

**Comment by neelesh17** [comment]
hey @sashadev-sky, i am still unclear about what is it we are trying to achive in this  issue.Would you help me out please.

**Comment by sashadev-sky** [comment]
@neelesh17 Currently the tooltips look like this: 

![image](https://user-images.githubusercontent.com/41092741/79805141-7c49bf00-8333-11ea-9abe-488e33071215.png)

And we were thinking to make them look nicer, closer to this: (this screenshot is from a toolbar in a different PL project, only the tooltip is relevant)

![image](https://user-images.githubusercontent.com/24359/43849016-5b74fbfc-9b02-11e8-81fe-0d70d51a6a8e.png)



but also open to other designs if theres a specific one you like and want to implement

**Comment by vanithaak** [comment]
> Hi @roshniRam are you still interested in this at all?
> 
> I think we may be able to remove the `tooltip` attribute here and just build in the Bootstrap-style tooltips into the `span`:
> 
> https://github.com/publiclab/Leaflet.DistortableImage/blob/90c977770658453d57233e1a150159cf7a38b1cb/src/edit/DistortableImage.EditToolbar.js#L14-L16
> 
> It would require calling the `tooltip` bootstrap javascript function here, too:
> 
> https://getbootstrap.com/docs/3.3/javascript/#tooltips

Hi @jywarren , @sashadev-sky , @TildaDares  I think the file "[DistortableImage.EditToolbar.js](https://github.com/publiclab/Leaflet.DistortableImage/blob/90c977770658453d57233e1a150159cf7a38b1cb/src/edit/DistortableImage.EditToolbar.js#L14-L16)" inside edit folder isn't their anymore. Please correct me if I'm wrong. Thanks!!

**Comment by jywarren** [comment]
Hi, good point! Do you think the code lives here now? 

https://github.com/publiclab/Leaflet.DistortableImage/blob/8b7211174c8b9607a039cf6bd9f091ec0aa4f5fc/src/edit/actions/EditAction.js#L10-L13

**Comment by vanithaak** [comment]
yes, I thought so, but ...

 - I tried adding html tags inside the html key, so that I can add my bootstrap attributes - didn't work
 - tried removing "title" attribute and set bootstrap attributes for tooltip with vanilla js - didn't work

What I'm able to do is remove the value of "title" attribute when I remove the tooltip key from the code.

More HINTS ! pretty please !
@jywarren
